### ISSUE TITLE
Plot layout sizes calculated wrong

### ISSUE THREAD

**Issue by Dunkhan** [oxyplot/oxyplot-sharpdx#8]
### Steps to reproduce
Add a comment if you have any issues reproducing this and I will try again.

All margins and padding were set to 0 in this example on model creation.

1. View any plot original renderer and the SharpDX renderer
2. Observe that the margins are significantly different

Platform: WPF Win10
.NET version: 4.5.2

### Expected behaviour

The margin calculations should not be different when using a different renderer

### Actual behaviour

There is a large empty space at the top of the plot. It looks as though the space for the title is being included even though there is no title shown.

SharpDX
![marginnodx](https://user-images.githubusercontent.com/4968631/36430766-3cba5904-1656-11e8-8906-7c9659c484a6.PNG)

Original Renderer
![margindx](https://user-images.githubusercontent.com/4968631/36430767-3cd32da8-1656-11e8-936f-2485a6ee2229.PNG)

Edit: Removed example code. It was not necessary

**Comment by Dunkhan** [comment]
I was investigating what appeared to be a different issue in another plot and in desperation I tried setting some negative margins. This worked, meaning that the issue was in fact the same as this one. It also means that this one is not related to whether there is a title visible.
In this example setting the top margin to -48 fixed the issue, even for different resolutions and window sizes.
In the newly discovered example setting the margins to (25,-15,0,-120) was required to get the same look as the non-DX version. 
Now that I look at my other plots, there are actually a lot of them that have other, less noticeable margin changes. I will leave the original text of my report because I do not wish to remove information until I am sure it is not relevant, and until I have something concrete to replace it with.

For now it is at least clear that the margins are either being calculated incorrectly, or are being rendered differently.

**Comment by Dunkhan** [comment]
I made a screenshot of the example project, and then swapped the PlotView back to the old oxyplot one and took the same screenshot. The difference is fairly clear. Please bear in mind that although the margin difference shown here is the most common one, other offsets can occur. In the example from my last comment where the bottom gap was 120, the whole plot was squashed into a line along the top and was entirely unreadable. The title was not affected but the axis lines and labels were.

DX PlotView
![exampledx](https://user-images.githubusercontent.com/4968631/36439769-26dd68d4-166e-11e8-8ed3-ae9f2512002e.PNG)
Oxy PlotView
![examplenodx](https://user-images.githubusercontent.com/4968631/36439770-26f845b4-166e-11e8-8bcd-6996413d158e.PNG)

**Comment by Dunkhan** [comment]
The issue was that empty and null strings were being measured, returning the font height and 0 width. This was creating additional space wherever a text could be shown but was not.

The behaviour implemented in this PR is copied directly from the original oxyplot CanvasRenderContext https://github.com/oxyplot/oxyplot/blob/fe84b034c1105fd0cd679d8cedd88d9c20fde417/Source/OxyPlot.Wpf/CanvasRenderContext.cs
and simply returns OxySize.Empty for null and empty strings

There is still one plot in my application that has major margin issues, but because this fix fixes every other plot, including the sharpDX example project, it seems that this remaining one is a different issue with a different cause. I will therefore be creating a new issue, allowing this one to be closed once the PR is merged.
